src/test/java/seedu/address/logic/commands/DependencyCommandTest.java

```diff
@@ -96,18 +109,35 @@ public void executeUndoRedo_validIndexUnfilteredList_success() throws Exception
         DependencyCommand dependencyCommand = new DependencyCommand(INDEX_FIRST_TASK, INDEX_SECOND_TASK);
         Task newTask = DependencyCommand.createDependantTask(dependantTask, dependeeTask);
 
-        Model expectedModel = new ModelManager(model.getTaskManager(), new UserPrefs());
-        expectedModel.updateTask(dependantTask, newTask);
-        expectedModel.commitTaskManager();
+        ModelManager expectedModelAdd = new ModelManager(model.getTaskManager(), new UserPrefs());
+        expectedModelAdd.updateTask(dependantTask, newTask);
+        expectedModelAdd.commitTaskManager();
         // add dependency
         dependencyCommand.execute(model, commandHistory);
         // undo -> reverts task manager back to previous state and filtered task list to show all
         // tasks
-        expectedModel.undoTaskManager();
-        assertCommandSuccess(new UndoCommand(), model, commandHistory, UndoCommand.MESSAGE_SUCCESS, expectedModel);
+        expectedModelAdd.undoTaskManager();
+        assertCommandSuccess(new UndoCommand(), model, commandHistory, UndoCommand.MESSAGE_SUCCESS, expectedModelAdd);
         // redo -> same first task deleted again
```

Comment:
same dependency added again?

Reply:
Yes because now im testing whether I can "toggle off" the dependency

Reply:
I changed the comment